packages/hardhat/contracts/strategy/CrabStrategyV2.sol

```diff
@@ -193,15 +195,29 @@ contract CrabStrategyV2 is StrategyBase, StrategyFlashSwap, ReentrancyGuard, Own
         require(msg.sender == weth || msg.sender == address(powerTokenController), "Cannot receive eth");
     }
 
+    /**
+     * @notice Tranfer vault NFT to new contract
+     * @dev strategy cap is set to 0 to avoid future deposits.
+     */
+    function transferVault(address _newStrategy) external onlyOwner {
+        IShortPowerPerp(powerTokenController.shortPowerPerp()).safeTransferFrom(address(this), _newStrategy, vaultId);
+        _setStrategyCap(0);
+
+        emit VaultTransferred(_newStrategy, vaultId);
+    }
+
     /**
      * @notice owner can set the strategy cap in ETH collateral terms
      * @dev deposits are rejected if it would put the strategy above the cap amount
      * @dev strategy collateral can be above the cap amount due to hedging activities
      * @param _capAmount the maximum strategy collateral in ETH, checked on deposits
      */
     function setStrategyCap(uint256 _capAmount) external onlyOwner {
```

Comment:
Might be more simple to make this be a public function instead of having an external that calls internal function?  Get why you changed it, but public can have same result, probably with smaller bytecode and small gas differences (it may be slightly more expensive to call?)

I don't have strong opinions though, other thoughts from folks?  Its also possible we just keep in mind for future if we have bytecode size issues later on and can change it.

Reply:
The only reason I used `external` is that I read it's cheap. But yeah it's not frequently called function and doesn't necessarily have to be external. Will change!